src/main/kotlin/ua/marchenko/artauction/auth/service/AuthenticationServiceImpl.kt

```diff
@@ -1,48 +1,58 @@
 package ua.marchenko.artauction.auth.service
 
-import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.ReactiveAuthenticationManager
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
-import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.core.userdetails.ReactiveUserDetailsService
 import org.springframework.security.crypto.password.PasswordEncoder
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
 import ua.marchenko.artauction.auth.jwt.JwtService
 import ua.marchenko.artauction.auth.controller.dto.AuthenticationRequest
 import ua.marchenko.artauction.auth.controller.dto.AuthenticationResponse
 import ua.marchenko.artauction.auth.controller.dto.RegistrationRequest
 import ua.marchenko.artauction.auth.mapper.toMongo
 import ua.marchenko.artauction.user.exception.UserAlreadyExistsException
-import ua.marchenko.artauction.user.mapper.toResponse
 import ua.marchenko.artauction.user.repository.UserRepository
 
 @Service
 class AuthenticationServiceImpl(
-    private val authManager: AuthenticationManager,
-    private val userDetailsService: UserDetailsService,
+    private val reactiveAuthenticationManager: ReactiveAuthenticationManager,
+    private val userDetailsService: ReactiveUserDetailsService,
     private val jwtService: JwtService,
     private val userRepository: UserRepository,
     private val passwordEncoder: PasswordEncoder,
 ) : AuthenticationService {
 
-    override fun authentication(authenticationRequest: AuthenticationRequest): AuthenticationResponse {
-        authManager.authenticate(
+    override fun authentication(authenticationRequest: AuthenticationRequest): Mono<AuthenticationResponse> {
+        return reactiveAuthenticationManager.authenticate(
             UsernamePasswordAuthenticationToken(
                 authenticationRequest.email,
                 authenticationRequest.password
             )
-        )
-        val user = userDetailsService.loadUserByUsername(authenticationRequest.email)
-        val accessToken = jwtService.generate(user)
-        return AuthenticationResponse(
-            accessToken = accessToken,
-        )
+        ).flatMap {
+            userDetailsService.findByUsername(authenticationRequest.email)
+                .flatMap { user ->
+                    val accessToken = jwtService.generate(user)
+                    Mono.just(
+                        AuthenticationResponse(
+                            accessToken = accessToken,
+                        )
+                    )
+                }
+        }.onErrorResume {
+            Mono.error(RuntimeException("Invalid credentials"))
+        }
```

Comment:
```suggestion
        return reactiveAuthenticationManager.authenticate(
            UsernamePasswordAuthenticationToken(
                authenticationRequest.email,
                authenticationRequest.password
            )
        )
            .then(userDetailsService.findByUsername(authenticationRequest.email))
            .map { user -> jwtService.generate(user)}
            .map { accessToken -> AuthenticationResponse(accessToken) }
            .onErrorMap { RuntimeException("Invalid credentials") }
```

And also it might be good idea to create custom exception instead of **RuntimeException** (or use 
*org.springframework.security.authentication.BadCredentialsException**)